bin/sdk/gutenberg.js

```diff
@@ -13,6 +21,21 @@ exports.config = ( { argv: { inputDir, outputDir }, getBaseConfig } ) => {
 	const editorScript = path.join( inputDir, 'editor.js' );
 	const viewScript = path.join( inputDir, 'view.js' );
 
+	baseConfig.module.rules[ 0 ].use = baseConfig.module.rules[ 0 ].use.map( rule => {
```

Comment:
if anything, for now this is too magic and the hard-coded indices kind of make it impossible to infer _why_ it's doing what it's doing, making it hard to update and maintain.

in other words, yes let's find a better way 😄

Reply:
> how to best go about this?

as briefly discussed in Slack I think we'll discover that this extends beyond the SDK

Calypso will need something here since blocks don't import `React` either.

probably my guess is that we'll have a default rule in the base config for this and then pass an option in the config which can override the pragma values

```js
getBaseConfig( {
  jsxTransform: {
    element: 'wp.element.createElement',
    fragment: 'wp.element.Fragment',
  }
} )
```

not sure I want to directly encode the babel semantics in there but I'm not strongly opposed to it either.

we might also consider using some path-based option for this rule. by default I think `gutenberg/extensions` should get the transform, but maybe some SDK derivative has a file within its path that should also get it

```js
getBaseConfig( {
  jsxTransform: [ {
    include: /my-plugin\/gutenberg\/.*\.jsx/,
    element: 'wp.element.createElement',
    fragment: 'wp.element.Fragment',
  } ]
} )
```

I'm not saying we need to have all this solved now, but that's somewhat where I'm coming from when I think about it. I'll be exploring this in the base config in #27864 as well so hopefully we can coordinate on it. I don't want to take over this PR so I'll work separately there but my changes/explorations should be easy to change should we adopt all the behaviors in this one

Reply:
also note https://babeljs.io/docs/en/options#merging

I think we can just add this as an `override` and it won't be hard or complicated

Reply:
noteworthy is this…

https://babeljs.io/docs/en/options#entryoptions

> false indicates that an entry is entirely disabled. This can be useful in contexts where ordering is important, but a separate condition is needed to decide if something is enabled.
> …
> would enable the two plugin for files in src, but two would still execute between one and three.

Reply:
Thanks for the investigation and sharing!

I'll put this on hold for now, happy to let it be superseded by your work, or we can pick and choose what's useful 👍